internal/pkg/agent/application/upgrade/rollback.go

```diff
@@ -174,16 +174,21 @@ func cleanup(log *logger.Logger, topDirPath string, removeMarker, keepLogs bool,
 
 	dirPrefix := fmt.Sprintf("%s-", AgentName)
 
+	log.Infof("versioned homes to keep: %v", versionedHomesToKeep)
+
+	var cumulativeError error
 	relativeHomePaths := make([]string, len(versionedHomesToKeep))
 	for i, h := range versionedHomesToKeep {
-		relHomePath, err := filepath.Rel("data", h)
+		relHomePath, err := filepath.Rel(dataDirPath, filepath.Join(topDirPath, h))
 		if err != nil {
-			return fmt.Errorf("extracting elastic-agent path relative to data directory from %s: %w", h, err)
+			cumulativeError = goerrors.Join(cumulativeError, fmt.Errorf("extracting elastic-agent path relative to data directory from %s: %w", h, err))
+			continue
```

Comment:
Looks like we were failing fast in the cleanup before but we are no longer doing that.  Why is that?

Reply:
This error can only happen if the watcher is passed some wrong paths in the list of versionedHomes to preserve during cleanup phase. 
If such an error occurs:
- (failing fast) -> Cleanup stops entirely, watcher exits, leaving obsolete installs on disk
- (accumulating the error) -> Cleanup keeps going and the entry causing the error is ignored, cleaning up everything that does not match entries in `relativeHomePaths`.

I was thinking if this should be a show stopper or if the watcher cleanup should keep going in best effort mode, I switched to accumulating the error and keep going but the previous behaviour is not wrong, I guess it depends on expectations.

What do you think should be the correct approach here ?